azerothcore/azerothcore-wotlk
# feat(CORE/Config): Added Pet experience rate to config

## CHANGES PROPOSED:
- Added Rate.XP.Pet to allow to change the rate at which pets gain experience
- Added Rate.Pet.LevelXP to change the required experience to level up the pet


## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3687



## TESTS PERFORMED:
- Captured pet, with default values, checked that it levels up normally
- Changed the rating to a big value like 900 killed enemies and confirm pet leveled up instantly multiple levels 



## HOW TO TEST THE CHANGES:

1. Tame a new pet (/cast tame beast)
2. Kill some enemies and verify pet levels up normally
3. Change rating to a big value
4. .reload config
5. Kill an enemy and confirm pet leveled up very fast
PS: In case needed required experience and current experience for the pet can be obtained by writing in chat: /dump GetPetExperience()


## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
